src/objects/zcl_abapgit_object_pdts.clas.testclasses.abap

```diff
@@ -0,0 +1,446 @@
+    add_line( |    </CONTAINER>| ).
+    add_line( |   </CONTAINER>| ).
+
+  ENDMETHOD.
+
+
+  METHOD create_input_xml.
+    generate( ).
+    CREATE OBJECT ri_result TYPE zcl_abapgit_xml_input
+      EXPORTING
+        iv_xml = mv_xml.
+  ENDMETHOD.
+
+
+  METHOD get_input_xml.
+    rv_result = me->mv_xml.
+  ENDMETHOD.
+
+ENDCLASS.
+
+
+CLASS ltc_turnaround_test DEFINITION FINAL FOR TESTING
+  DURATION LONG
+  RISK LEVEL CRITICAL.
+
+  PRIVATE SECTION.
+
+    DATA mo_mock TYPE REF TO ltd_mock.
+    DATA mi_output_xml TYPE REF TO zif_abapgit_xml_output.
+    DATA mo_cut    TYPE REF TO zif_abapgit_object.
+
+    CLASS-METHODS class_setup.
+    CLASS-METHODS task_exists RETURNING VALUE(rv_result) TYPE abap_bool.
+
+    METHODS setup.
+
+    METHODS create_task RAISING zcx_abapgit_exception.
+    METHODS serialize_task RETURNING VALUE(rv_result) TYPE string
+                           RAISING   zcx_abapgit_exception.
+    METHODS clean_xml IMPORTING iv_xml           TYPE string
+                      RETURNING VALUE(rv_result) TYPE string.
+    METHODS delete_task
+      RAISING
+        zcx_abapgit_exception.
+
+    METHODS output_matches_input FOR TESTING RAISING cx_static_check.
+
+ENDCLASS.
+
+
+
+CLASS ltc_turnaround_test IMPLEMENTATION.
+
+
+  METHOD class_setup.
+
+    IF task_exists( ) = abap_true.
+      cl_abap_unit_assert=>fail( msg   = |Test task { ltd_mock=>mc_task_id } already exists|
+                                 level = if_aunit_constants=>fatal
+                                 quit  = if_aunit_constants=>class ).
+    ENDIF.
+
+  ENDMETHOD.
+
+
+  METHOD setup.
+    DATA  ls_item   TYPE zif_abapgit_definitions=>ty_item.
+
+    ls_item-obj_type = 'PDTS'.
+    ls_item-obj_name = ltd_mock=>mc_task_id.
+
+    TRY.
+        CREATE OBJECT mo_cut TYPE zcl_abapgit_object_pdts
+          EXPORTING
+            is_item     = ls_item
+            iv_language = sy-langu.
+      CATCH zcx_abapgit_exception.
+        cl_abap_unit_assert=>fail( ).
+    ENDTRY.
+
+    CREATE OBJECT mo_mock.
+
+  ENDMETHOD.
+
+
+  METHOD output_matches_input.
+
+    DATA lv_output TYPE string.
+
+    create_task( ).
+    cl_abap_unit_assert=>assert_equals( act = mo_cut->changed_by( )
+                                        exp = sy-uname ).
+    lv_output = serialize_task( ).
+    lv_output = clean_xml( lv_output ).
+
+    cl_abap_unit_assert=>assert_equals( act = lv_output
+                                        exp = mo_mock->get_input_xml( ) ).
+    delete_task( ).
+
+  ENDMETHOD.
+
+
+  METHOD clean_xml.
+    rv_result = substring_from( val = iv_xml
+                                sub = '<' ).
+  ENDMETHOD.
+
+
+  METHOD create_task.
+
+    DATA lo_input_xml TYPE REF TO zif_abapgit_xml_input.
+
+    DATA: lv_step TYPE zif_abapgit_definitions=>ty_deserialization_step,
+          li_log  TYPE REF TO zif_abapgit_log.              "#EC NEEDED
+
+    lo_input_xml = mo_mock->create_input_xml( ).
+
+    mo_cut->deserialize( iv_package = '$TMP'
```

Comment:
does this make changes to the system? I'd like to keep the user's systems clean, if they accidently runs the unit tests

Reply:
Yes, it creates a task from a dummy repo, serialises it, compares it with the source, deletes it.

This is why I marked it as DANGEROUS, and they would need to be running a fully installed abapGit instead of the standalone version.

Reply:
Added check for "Enable critical tests" config setting.

Reply:
(We should centralise this, thinking of adding a global test class for stuff like this)

Reply:
its centralized as https://github.com/abapGit-tests via https://github.com/abapGit/CI

there are very few places in the main codebase containing critical tests

Reply:
I meant centralise this and similar checks - for now I just duplicated code from `zcl_abapgit_objects`/`ltcl_dangerous`, but this means it should move somewhere central. 
IMHO it *must* be possible to run unit tests locally without waiting for CI. During refactoring I might run a test every few seconds. Delete lines, test, rename some stuff, test, do another quick fix, test. This is only feasible with sub-second test runs.

Reply:
https://github.com/abapGit/abapGit/issues/3953 opened

https://github.com/abapGit/CI can already run locally